From 5a92c7500d94fcd11b47b77973308b367bf32abd Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Fri, 7 Aug 2015 15:23:17 -0700
Subject: [PATCH 24/43] libparted: Fix misspelling in hfs exception string

---
 libparted/fs/r/hfs/journal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libparted/fs/r/hfs/journal.c b/libparted/fs/r/hfs/journal.c
index 9080e14..069a8bd 100644
--- a/libparted/fs/r/hfs/journal.c
+++ b/libparted/fs/r/hfs/journal.c
@@ -257,7 +257,7 @@ hfsj_replay_journal(PedFileSystem* fs)
 			PED_EXCEPTION_NO_FEATURE,
 			PED_EXCEPTION_CANCEL,
 			_("Journal stored outside of the volume are "
-			  "not supported.  Try to desactivate the "
+			  "not supported.  Try to deactivate the "
 			  "journal and run Parted again."));
 		return 0;
 	}
-- 
1.9.5

