From 3447264ba9fedf236da92b2199a2b4823b773cf5 Mon Sep 17 00:00:00 2001
From: Phillip Susi <psusi@ubuntu.com>
Date: Thu, 14 Apr 2016 20:58:00 -0400
Subject: [PATCH 41/43] Add NEWS entry for fat resize fix

---
 NEWS | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/NEWS b/NEWS
index 0573ae7..cf7b39c 100644
--- a/NEWS
+++ b/NEWS
@@ -4,6 +4,9 @@ GNU parted NEWS                                    -*- outline -*-
 
 ** Bug Fixes
 
+  Fix fat resize to preverve boot code, and thus not render the
+  filesystem unreconized by Windows.
+
   Fix rescue command: the rescue command often failed to find
   filesystems due to leaving on cylinder alignment.
 
-- 
1.9.5

