From 2a13cc944e827dd23d2ef01a69c731fb017eb48f Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Tue, 4 Oct 2016 16:11:25 -0700
Subject: [PATCH 56/75] tests: t3310-flags.sh skip pc98 when sector size != 512

---
 tests/t3310-flags.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/t3310-flags.sh b/tests/t3310-flags.sh
index e97c3b9..10ac50d 100644
--- a/tests/t3310-flags.sh
+++ b/tests/t3310-flags.sh
@@ -49,6 +49,8 @@ for table_type in aix amiga bsd dvh gpt mac msdos pc98 sun loop; do
     msdos) primary_or_name='primary'
            ;;
     pc98)  primary_or_name='PTNNAME'
+           # pc98 only supports 512b sectors
+           [ $ss -ne 512 ] && continue
            ;;
     sun)   primary_or_name=''
            ;;
-- 
2.8.3

