From e366a80d9a3d972a6a1b3c7b2c9b4a77d1f52e0c Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Tue, 6 Jun 2017 11:05:31 -0700
Subject: [PATCH 90/94] libparted: Fix udev cookie leak in _dm_resize_partition

Add a NEWS entry.
---
 NEWS | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/NEWS b/NEWS
index b6aeb89..bb15212 100644
--- a/NEWS
+++ b/NEWS
@@ -5,6 +5,9 @@ GNU parted NEWS                                    -*- outline -*-
 * Noteworthy changes in release ?.? (????-??-??) [?]
 
 ** Bug Fixes
+
+  Fix a udev cookie leak when using resizepart on device-mapper devices.
+
   Fix a gettext crash/error sometimes when using localized languages.
 
   Fix fat resize to preverve boot code, and thus not render the
-- 
2.8.3

