diff -Naur parted-3.2/libparted/arch/linux.c parted-3.2.tpg/libparted/arch/linux.c
--- parted-3.2/libparted/arch/linux.c	2018-04-24 13:13:46.117759000 +0000
+++ parted-3.2.tpg/libparted/arch/linux.c	2018-04-24 13:22:22.937680579 +0000
@@ -297,6 +297,8 @@
 static int _device_open (PedDevice* dev, int flags);
 static int _device_open_ro (PedDevice* dev);
 static int _device_close (PedDevice* dev);
+static int linux_close (PedDevice* dev);
+static void _flush_cache (PedDevice* dev);
 
 static int
 _read_fd (int fd, char **buf)
@@ -450,6 +452,24 @@
         return _major_type_in_devices (major, "blkext");
 }
 
+static void
+_ensure_read_write (PedDevice *dev)
+{
+       PED_ASSERT (dev != NULL);
+       LinuxSpecific* arch_specific = LINUX_SPECIFIC (dev);
+
+       if (arch_specific->rw)
+              return;
+
+       if (!linux_close(dev))
+              return;
+
+       if (!_device_open (dev, RW_MODE))
+             return;
+
+       _flush_cache (dev);
+}
+
 #ifdef ENABLE_DEVICE_MAPPER
 static int
 _dm_task_run_wait (struct dm_task *task, uint32_t cookie)
@@ -1639,6 +1659,7 @@
 
         if (dev->read_only || dev->type == PED_DEVICE_RAM)
                 return;
+        _ensure_read_write (dev);
         dev->dirty = 0;
 
         ioctl (arch_specific->fd, BLKFLSBUF);
@@ -1682,7 +1703,7 @@
 static int
 linux_open (PedDevice* dev)
 {
-    return _device_open (dev, RW_MODE);
+    return _device_open (dev, RD_MODE);
 }
 
 static int
@@ -1720,10 +1741,12 @@
                 }
         } else {
                 dev->read_only = 0;
+                if (flags == WR_MODE || flags == RW_MODE)
+                       arch_specific->rw = 1;
+               else
+                       arch_specific->rw = 0;
         }
 
-        _flush_cache (dev);
-
         return 1;
 }
 
@@ -1802,6 +1825,7 @@
         PED_ASSERT (dev->sector_size % PED_SECTOR_SIZE_DEFAULT == 0);
         PED_ASSERT (dev != NULL);
         PED_ASSERT (!dev->external_mode);
+        _ensure_read_write (dev);
 
         arch_specific = LINUX_SPECIFIC (dev);
 
@@ -2003,6 +2027,7 @@
                         return 1;
         }
 
+        _ensure_read_write (dev);
         if (_get_linux_version() < KERNEL_VERSION (2,6,0)) {
                 /* Kludge.  This is necessary to read/write the last
                    block of an odd-sized disk, until Linux 2.5.x kernel fixes.
@@ -2532,6 +2557,7 @@
         LinuxSpecific*          arch_specific = LINUX_SPECIFIC (dev);
         struct blkpg_ioctl_arg  ioctl_arg;
 
+        _ensure_read_write (dev);
         ioctl_arg.op = op;
         ioctl_arg.flags = 0;
         ioctl_arg.datalen = sizeof (struct blkpg_partition);
diff -Naur parted-3.2/libparted/arch/linux.h parted-3.2.tpg/libparted/arch/linux.h
--- parted-3.2/libparted/arch/linux.h	2014-06-15 18:17:43.000000000 +0000
+++ parted-3.2.tpg/libparted/arch/linux.h	2018-04-24 13:15:14.032089148 +0000
@@ -30,6 +30,7 @@
 	int	fd;
 	int	major;
 	int	minor;
+	int	rw;
 	char*	dmtype;         /**< device map target type */
 #if defined __s390__ || defined __s390x__
 	unsigned int real_sector_size;
