diff -Naur parted-3.3/libparted/fsresize/fat/fat.c parted-3.3.tpg/libparted/fsresize/fat/fat.c
--- parted-3.3/libparted/fsresize/fat/fat.c	2019-10-10 22:58:51.000000000 +0000
+++ parted-3.3.tpg/libparted/fsresize/fat/fat.c	2019-10-15 20:18:35.806278542 +0000
@@ -21,7 +21,7 @@
 
 #include "fat.h"
 #include "calc.h"
-#include "../../../labels/misc.h"
+#include "../../labels/misc.h"
 
 PedFileSystem*
 fat_alloc (const PedGeometry* geom)
