diff -Naur parted-3.2/configure.ac parted-3.2.tpg/configure.ac
--- parted-3.2/configure.ac	2014-06-15 18:17:43.000000000 +0000
+++ parted-3.2.tpg/configure.ac	2016-06-29 20:44:43.063631221 +0000
@@ -165,7 +165,8 @@
 
 dnl Check for programs.
 AC_ISC_POSIX
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
 AC_PROG_GCC_TRADITIONAL
 AM_PROG_CC_C_O
 
@@ -377,7 +378,7 @@
 if test "$with_readline" = yes; then
 	OLD_LIBS="$LIBS"
 	LIBS=""
-	AC_SEARCH_LIBS([tgetent], [tinfo ncurses curses termcap termlib],
+	AC_SEARCH_LIBS([tgetent], [tinfo ncursesw curses termcap termlib],
 		[PARTED_LIBS="$PARTED_LIBS $LIBS"],
 		[AC_MSG_ERROR(dnl
 [termcap could not be found which is required for the
