--- parted-3.2/libparted/fs/Makefile.in.omv~	2015-01-27 21:27:15.535224747 +0100
+++ parted-3.2/libparted/fs/Makefile.in	2015-01-27 21:27:25.565306987 +0100
@@ -213,7 +213,7 @@ AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
 am__v_lt_0 = --silent
 am__v_lt_1 = 
-libparted_fs_resize_la_LIBADD =
+libparted_fs_resize_la_LIBADD = -luuid
 am_libparted_fs_resize_la_OBJECTS = r/filesys.lo r/fat/bootsector.lo \
 	r/fat/calc.lo r/fat/clstdup.lo r/fat/context.lo r/fat/count.lo \
 	r/fat/fat.lo r/fat/fatio.lo r/fat/resize.lo r/fat/table.lo \
