diff -Naur parted-3.4/libparted/Makefile.am parted-3.4.tpg/libparted/Makefile.am
--- parted-3.4/libparted/Makefile.am	2021-01-25 19:49:56.000000000 +0000
+++ parted-3.4.tpg/libparted/Makefile.am	2021-01-28 16:17:32.973000000 +0000
@@ -13,7 +13,7 @@
 
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 
-SUBDIRS       = labels fs . $(SUBDIRS_CHECK)
+SUBDIRS       = labels . fs $(SUBDIRS_CHECK)
 
 
 partedincludedir = \
diff -Naur parted-3.4/libparted/fs/Makefile.am parted-3.4.tpg/libparted/fs/Makefile.am
--- parted-3.4/libparted/fs/Makefile.am	2021-01-25 19:49:56.000000000 +0000
+++ parted-3.4.tpg/libparted/fs/Makefile.am	2021-01-28 14:09:10.318000000 +0000
@@ -74,6 +74,7 @@
   -version-info $(CURRENT):$(REVISION):$(AGE)
 EXTRA_DIST += fsresize.sym
 libparted_fs_resize_la_DEPENDENCIES = $(sym_file)
+libparted_fs_resize_la_LIBADD = $(top_builddir)/libparted/libparted.la
 
 libparted_fs_resize_la_SOURCES = \
   r/filesys.c			\
